src/py42/sdk/__init__.py

```diff
@@ -72,14 +74,17 @@ def from_local_account(cls, host_address, username, password):
                 console.us.code42.com
             username (str): The username of the authenticating account.
             password (str): The password of the authenticating account.
-
+            totp (str): The time-based one-time password of the authenticating account. Include only
+                if the account uses Code42's two-factor authentication.
         Returns:
             :class:`py42.sdk.SDKClient`
         """
         basic_auth = None
         if username and password:
             basic_auth = HTTPBasicAuth(username, password)
         auth_connection = Connection.from_host_address(host_address, auth=basic_auth)
+        if totp:
+            auth_connection._headers.update({"totp-auth": str(totp)})
```

Comment:
Might be worth adding a `add_header()` method to `py42.services._connection.Connection` class so we don't have to refer to internal attribute `_headers` from outside the class.

Reply:
Added the method and updated the call in __init__.py.

Reply:
instead of doing this here, we can add an optional constructor param to `V3Auth` and use the `headers` keyword arg in the `get` call that requests the token.

Reply:
What if we had a new `C42RenewableAuth` TOTP subclass for this, that could take either a single token (and if the auth expired would not be able to auto-renew), but if given a callable that returned a new totp _would_ be able to auto-renew?

Reply:
Would it be a subclass of `C42RenewableAuth` or of `V3Auth`? Since we do want to use V3Auth regardless?

Reply:
@timabrmsn we don't need a new subclass, but I like the idea of changing the optional constructor param to be the callable you're talking about instead of the totp itself. This would allow the user to do something as simple as write a function that prompts for input _or_ something more complex like actually determining their totp from the secret key, and if you know that you will be needing 2fa this approach makes it so that you don't _have_ to catch `Py42MFAError`.

@ceciliastevens let me know if you have any questions about making that change.

Reply:
Makes sense to me. I've made the change that I think accomplishes this, if it looks good to you I'll go ahead and work on adding some tests next.

Reply:
oh wait just saw your latest comment. Give me a moment to revise out the add_header method